great_expectations/datasource/data_connector/configured_asset_file_path_data_connector.py

```diff
@@ -74,21 +75,20 @@ def _build_assets_from_config(self, config: Dict[str, dict]):
         for name, asset_config in config.items():
             if asset_config is None:
                 asset_config = {}
+            asset_config.update({"name": name})
             new_asset: Asset = self._build_asset_from_config(
-                name=name,
                 config=asset_config,
             )
             self.assets[name] = new_asset
 
-    def _build_asset_from_config(self, name: str, config: dict):
-        runtime_environment: dict = {"name": name, "data_connector": self}
+    def _build_asset_from_config(self, config: dict):
+        runtime_environment: dict = {"data_connector": self}
+        config = assetConfigSchema.load(config)
+        config = assetConfigSchema.dump(config)
```

Comment:
This might be a dumb question (non-blocking) - why is the config loaded then dumped?

Reply:
For schema validation (loaded to make sure that the supplied config passes the schema requirements; dumped because the instantiator takes a dict.